### PR TITLE
Add dprint fixer

### PR DESCRIPTION
**Pull request by myobie**
This is attempt number 3. See https://github.com/dense-analysis/ale/pulls?q=is%3Apr+is%3Aclosed+dprint

I have changed to use `stdin`. This is working well for me so far 💪

### CODE REVIEW

File: test/fixers/test_dprint_fixer_callback.vader [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Execute(The dprint callback should return the correct default values):
+  AssertFixer
+  \ {
+  \   'read_temporary_file': 1,
+  \   'command':
+  \     ale#Escape('dprint')
+  \     . ' ' . ale#Escape('')
+  \     . ' %t',
+  \ }
```
Comment (hsanson) [inline test/fixers/test_dprint_fixer_callback.vader:16]: Make sure this dictionary matches exactly what the fale#fixers#dprint#Fix() method returns:

```
    \ {
    \   'command': ale#Escape('dprint') . ' fmt --stdin %s',
    \ }
```

File: test/fixers/test_dprint_fixer_callback.vader [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  let g:ale_typescript_dprint_options = '-c dprint.json'
+
+  AssertFixer
+  \ {
+  \   'read_temporary_file': 1,
+  \   'command':
+  \     ale#Escape('dprint')
+  \     . ' ' . ale#Escape('-c dprint.json')
+  \     . ' %t',
+  \ }
```
Comment (hsanson) [inline test/fixers/test_dprint_fixer_callback.vader:28]: Same as previous test make sure the dictionary matches exactly what the fale#fixers#dprint#Fix() method returns:

```
    \ {
    \   'command': ale#Escape('dprint') . ' fmt -c dprint.json --stdin %s',
    \ }
```

Comment (myobie) [reply test/fixers/test_dprint_fixer_callback.vader:28]: Thanks. On it. Sorry to take some time, I saw the failures but haven't been able to address them yet. 💪

**Comment by myobie** [comment]
@hsanson sorry for noise in this PR (I had to get docker installed locally so I could run the tests). It's now ready to go. The problem I couldn't figure out was that, of course, `dprint` is not installed inside the test docker container. I added some extra tests and I feel good about how they look. Thanks for your patience.

**Review by hsanson — CHANGES_REQUESTED** [review]
Looking good, thanks for the work.  Would be great if you can add also help documentation to the PR:

- Add  `ale-dprint-options` section into `ale.txt` explaining the possible configuration variables.
- Add `ale-javascript-dprint` section to `ale-javascript.txt` which points to `ale-dprint-options`.  See ale-javascript-deno and ale-javascript-cspell for example.
- Repeat for every language supported by dprint (e.g. typescript, json, and markdown).

**Comment by myobie** [comment]
Thank you for your help getting this PR into shape. I've added all the languages which `dprint` supports with their `wasm` plugins. There are some process plugins, but they have a warning on their website and I think they are new. I'll revisit the ale docs in the future once I see how those non-`wasm` plugins turn out.

File: doc/ale.txt [hunk 3]
```diff
@@ -2597,7 +2597,53 @@ g:ale_cspell_use_global                               *g:ale_cspell_use_global*
...
 -------------------------------------------------------------------------------
-7.3. Options for languagetool                        *ale-languagetool-options*
+7.3. Options for dprint                                    *ale-dprint-options*
+
+`dprint` is a fixer for many file types, including: (java|type)script,
+json(c?), markdown, and more. See https://dprint.dev/plugins for an up-to-date
+list of supported plugins and their configuration options.
+
+The ale options for `dprint` are shared between all filetypes, so options can
+be configured only once.
```
Comment (hsanson) [inline doc/ale.txt:2607]: This is not entirely true. It is possible to add different configurations per filetype using ftplugin configurations or filetype autocommands, etc.

Comment (myobie) [reply doc/ale.txt:2607]: Oh! I see. I copied `cspell` a bit too closely. I'll change this right now.

**Review by hsanson — CHANGES_REQUESTED** [review]
Thanks for the extensive documentation. Just one small detail and this should be ready to merge.

**Review by hsanson — APPROVED** [review]
Thanks, good contribution and well documented.

**Pull request merged**